split filterstring in comma separated parts

kdschmidt1 opened the pull request:
(otherwise filter definitions as documented like  FILTER=
'$HDG,$HDM,$HDT,$VHW' in plugins are not working)

wellenvogel commented:
we cannot do it like this.
The API allows to provide a list as filter parameter - see https://github.com/wellenvogel/avnav/blob/master/server/avnav_api.py#L98
Most probably the description is not really clear (currently only a single filter or a list of filters is working).
And as at least on the html page there is "like in avnav_server.xml" we should split the parameter if it is no list...
Finally we could also split each element of a list - but maybe this would be some overkill.

kdschmidt1 commented:
Hallo andreas,

aufgefallen ist das nur dadurch, dass du exakt diese filterdefinition FILTER= '$HDG,$HDM,$HDT,$VHW' in deiner änderung im more-nmea-plugin verwendet hast, die aber so nicht funktioniert.

Ich habe das zunächst gendert in FILTER= ['$HDG','$HDM','$HDT','$VHW'], das funktioniert dann.

Bei einer änderung von pluginhandler in 

  def fetchFromQueue(self, sequence, number=10,includeSource=False,waitTime=0.5,filter=None):

    if filter is not None:

      if not (isinstance(filter,list)):

        filter=filter.split(',')

    return self.queue.fetchFromHistory(sequence,number,includeSource=includeSource,waitTime=waitTime,nmeafilter=filter)

(anders als in meinem pull request) funktionieren dann aber sogar beide Varianten.

Gruss 

kd

 

________________________

Dr.-Ing. Klaus Dieter Schmidt

Hauptstrasse 52A

D-78269 Volkertshausen

***@***.***

*+49 171 36 19 275

 

 

 

Von: Andreas Vogel ***@***.***> 
Gesendet: Freitag, 14. Mai 2021 12:01
An: wellenvogel/avnav ***@***.***>
Cc: kdschmidt1 ***@***.***>; Author ***@***.***>
Betreff: Re: [wellenvogel/avnav] split filterstring in comma separated parts (#131)

 

we cannot do it like this.
The API allows to provide a list as filter parameter - see https://github.com/wellenvogel/avnav/blob/master/server/avnav_api.py#L98
Most probably the description is not really clear (currently only a single filter or a list of filters is working).
And as at least on the html page there is "like in avnav_server.xml" we should split the parameter if it is no list...
Finally we could also split each element of a list - but maybe this would be some overkill.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/wellenvogel/avnav/pull/131#issuecomment-841144487> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ANNAQYT2JKD57TZVQYVX6NLTNTYD5ANCNFSM444B6BDQ> .

wellenvogel commented:
genau so machen wir es...
